silkworm/core/chain/config.hpp

```diff
@@ -119,95 +127,19 @@ struct ChainConfig {
-    .terminal_total_difficulty = intx::from_string<intx::uint256>("58750000000000000000000"),
-    .shanghai_time = 1681338455,
-};
+constinit extern const ChainConfig kMainnetConfig;
 
 inline constexpr evmc::bytes32 kGoerliGenesisHash{0xbf7e331f7f7c1dd2e05159666b3bf8bc7a8a3a9eb1d518969eab529dd9b88c1a_bytes32};
-inline constexpr ChainConfig kGoerliConfig{
-    .chain_id = 5,
-    .protocol_rule_set = protocol::RuleSetType::kClique,
-    .homestead_block = 0,
-    .tangerine_whistle_block = 0,
-    .spurious_dragon_block = 0,
-    .byzantium_block = 0,
-    .constantinople_block = 0,
-    .petersburg_block = 0,
-    .istanbul_block = 1'561'651,
-    .berlin_block = 4'460'644,
-    .london_block = 5'062'605,
-    .terminal_total_difficulty = 10790000,
-    .shanghai_time = 1678832736,
-};
+constinit extern const ChainConfig kGoerliConfig;
 
 inline constexpr evmc::bytes32 kSepoliaGenesisHash{0x25a5cc106eea7138acab33231d7160d69cb777ee0c2c553fcddf5138993e6dd9_bytes32};
-inline constexpr ChainConfig kSepoliaConfig{
-    .chain_id = 11155111,
-    .protocol_rule_set = protocol::RuleSetType::kEthash,
-    .homestead_block = 0,
-    .tangerine_whistle_block = 0,
-    .spurious_dragon_block = 0,
-    .byzantium_block = 0,
-    .constantinople_block = 0,
-    .petersburg_block = 0,
-    .istanbul_block = 0,
-    .muir_glacier_block = 0,
-    .berlin_block = 0,
-    .london_block = 0,
-    .terminal_total_difficulty = 17000000000000000,
-    .merge_netsplit_block = 1'735'371,
-    .shanghai_time = 1677557088,
-};
+constinit extern const ChainConfig kSepoliaConfig;
 
 inline constexpr evmc::bytes32 kPolygonGenesisHash{0xa9c28ce2141b56c474f1dc504bee9b01eb1bd7d1a507580d5519d4437a97de1b_bytes32};
-inline constexpr ChainConfig kPolygonConfig{
-    .chain_id = 137,
-    .protocol_rule_set = protocol::RuleSetType::kBor,
-    .homestead_block = 0,
-    .tangerine_whistle_block = 0,
-    .spurious_dragon_block = 0,
-    .byzantium_block = 0,
-    .constantinople_block = 0,
-    .petersburg_block = 0,
-    .istanbul_block = 3'395'000,
-    .muir_glacier_block = 3'395'000,
-    .berlin_block = 14'750'000,
-    .london_block = 23'850'000,
-    .eip1559_fee_collector = 0x70bca57f4579f58670ab2d18ef16e02c17553c38_address,
-};
+extern const ChainConfig kPolygonConfig;
 
 inline constexpr evmc::bytes32 kMumbaiGenesisHash{0x7b66506a9ebdbf30d32b43c5f15a3b1216269a1ec3a75aa3182b86176a2b1ca7_bytes32};
-inline constexpr ChainConfig kMumbaiConfig{
-    .chain_id = 80001,
-    .protocol_rule_set = protocol::RuleSetType::kBor,
-    .homestead_block = 0,
-    .tangerine_whistle_block = 0,
-    .spurious_dragon_block = 0,
-    .byzantium_block = 0,
-    .constantinople_block = 0,
-    .petersburg_block = 0,
-    .istanbul_block = 2'722'000,
-    .muir_glacier_block = 2'722'000,
-    .berlin_block = 13'996'000,
-    .london_block = 22'640'000,
-    .eip1559_fee_collector = 0x70bca57f4579f58670ab2d18ef16e02c17553c38_address,
-};
+extern const ChainConfig kMumbaiConfig;
```

Comment:
Why not `constinit` for `kPolygonConfig` and `kMumbaiConfig`?

Reply:
> Why not `constinit` for `kPolygonConfig` and `kMumbaiConfig`?

Because it's not compatible with `std::map` unfortunately. Strictly speaking, using `std::map` in static variables is in violation of the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables), but I'm thinking that it's not that big a deal.